service/middleware.go

```diff
@@ -71,3 +73,16 @@ func Middleware(inner http.Handler, env *Env) http.Handler {
 		inner.ServeHTTP(w, r)
 	})
 }
+
+// CORSMiddleware handles the header options for cross-origin requests (used in development only)
+func CORSMiddleware() func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		headers := handlers.AllowedHeaders([]string{"Content-Type", "Content-Length", "Accept-Encoding", "X-CSRF-Token", "X-Requested-With", "Origin"})
+		origins := handlers.AllowedOrigins([]string{"http://localhost:3000"})
+		methods := handlers.AllowedMethods([]string{"GET", "HEAD", "POST", "PUT", "OPTIONS"})
```

Comment:
With dev mode you can also get OPTIONS - that's generated by a cross-origin request (CORS). That said, I spent a *lot* of time trying to get access from port 3000 working and gave up in the end!